credoai/prism/prism.py

```diff
@@ -0,0 +1,86 @@
+from typing import List
+
+from credoai.lens import Lens
+from credoai.prism.task import Task
+from credoai.utils import ValidationError, global_logger
+from credoai.utils.common import NotRunError
+
+
+class Prism:
+    """
+    **Experimental**
+
+    Orchestrates the run of complex operations (Tasks) on
+    an arbitrary amount of Lens objects.
+
+    Parameters
+    ----------
+    lenses : List[Lens]
+        A list of Lens objects. The only requirement is for the Lens objects to
+        be instantiated with their necessary artifacts. One or multiple Lens objects
+        can be provided, each Task will validate that the amount of objects provided
+        is suitable for its requirement.
+    task : Task
```

Comment:
You said tasks work as evaluators, but here I see I can only pass one task.

Reply:
Yeah I consciously constrained the power of Prism, task as we imagined them would be of 2 types:

1. Calculations on Lens results, produce numerical results (compare for example)
2. Operations on the instances, for example creating new ones based on sampling of datasets

I imagine that at the end of a Prism run some form of numerical results need to be produced. 
That being the case, a Type 2 task would require another task of type 1 to be run after, so that numerical output can be produced. This would require to enforce order of execution (or to add task dependency on other tasks).

I decided to avoid the complication by allowing for one task to call another one. So for example an hypothetical cross-validation task would:

1. Create the Lens instances by splitting dataset appropriately
2. Run `compare` to produce numerical results

ATM I don't imagine the need to run super complex sequences of tasks, this might be a reflection on the fact that we don't have an actual use case for Prism yet.